lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -190,6 +212,26 @@ public void close()
         bufferedBytes = 0;
     }
 
+    public void validate(Function<ParquetWriteValidation, ConnectorPageSource> input)
+            throws IOException
+    {
+        checkState(validationBuilder != null, "validation is not enabled");
+        ParquetWriteValidation writeValidation = validationBuilder.build();
+        try (ConnectorPageSource pageSource = input.apply(writeValidation)) {
+            for (Page page = pageSource.getNextPage(); page != null; page = pageSource.getNextPage()) {
```

Comment:
can it happen that `page` will be `null` but page source is not finished? Should we check `io.trino.spi.connector.ConnectorPageSource#isFinished` instead